clang/lib/Serialization/ASTReader.cpp

```diff
@@ -918,7 +918,7 @@ ASTSelectorLookupTrait::ReadKey(const unsigned char* d, unsigned) {
   SelectorTable &SelTable = Reader.getContext().Selectors;
   unsigned N = endian::readNext<uint16_t, llvm::endianness::little>(d);
   const IdentifierInfo *FirstII = Reader.getLocalIdentifier(
-      F, endian::readNext<uint32_t, llvm::endianness::little>(d));
+      F, endian::readNext<IdentifierID, llvm::endianness::little>(d));
```

Comment:
I think having some kind of `static_assert` that `IdentifierID` is an integral type would be helpful. Maybe that'd be useful even within `endian::readNext()`?

Reply:
You're right, it would fail when looking for a `getSwappedBytes()` overload.